src/main/java/org/breedinginsight/services/ProgramUserService.java

```diff
@@ -274,6 +275,24 @@ public boolean existsAndActive(UUID programId, UUID userId) {
         return programUserDao.existsAndActive(programId, userId);
     }
 
+    /**
+     * Get a program user by programId and userId if it is in the Experimental Collaborator role.
+     * @param programId the program ID.
+     * @param userId the user ID.
+     * @return an Optional that unwraps to a program user if it is an Experimental Collaborator, Optional.empty() otherwise.
+     */
+    public Optional<ProgramUser> getIfExperimentalCollaborator(UUID programId, UUID userId) throws DoesNotExistException {
+        Optional<ProgramUser> programUser = getProgramUserbyId(programId, userId);
+        if (programUser.isEmpty()) {
+            throw new DoesNotExistException("Program User does not exist");
+        }
+        boolean isExperimentalCollaborator = programUser.get().getRoles().stream().anyMatch(x -> ProgramSecuredRole.getEnum(x.getDomain()).equals(ProgramSecuredRole.EXPERIMENTAL_COLLABORATOR));
+        if (isExperimentalCollaborator) {
+            return programUser;
+        }
+        return Optional.empty();
+    }
```

Comment:
Could refactor here using Optional methods
```suggestion
       public Optional<ProgramUser> getIfExperimentalCollaborator(UUID programId, UUID userId) throws DoesNotExistException {
        return getProgramUserbyId(programId, userId)
                .filter(programUser -> programUser.getRoles().stream()
                        .anyMatch(x -> ProgramSecuredRole.getEnum(x.getDomain()).equals(ProgramSecuredRole.EXPERIMENTAL_COLLABORATOR)))
                .or(() -> {
                    throw new RuntimeException(new DoesNotExistException("Program User does not exist"));
                });
    }
```

Reply:
Thank you for the suggestion. I tested it locally and it caused tests to fail because returning `Optional.empty()` is an important case which conceptually indicates that the program user exists, but is not in the "Experimental Collaborator" role.